patch/llpcPatchBufferOp.cpp

```diff
@@ -1595,192 +1595,159 @@ Value* PatchBufferOp::ReplaceLoad(
     }
 
     SmallVector<Value*, 8> partLoads;
+    Type* smallestType;
+    uint32_t smallestByteSize = UINT32_MAX;
 
     uint32_t remainingBytes = bytesToLoad;
 
-    // If the alignment is at least 4, we can use the most efficient dword loads.
-    if (alignment >= 4)
+    while (remainingBytes > 0)
     {
-        while (remainingBytes >= 4)
-        {
-            const uint32_t offset = bytesToLoad - remainingBytes;
-            Value* const pOffset = (offset == 0) ?
-                                   pBaseIndex :
-                                   m_pBuilder->CreateAdd(pBaseIndex, m_pBuilder->getInt32(offset));
+        const uint32_t offset = bytesToLoad - remainingBytes;
+        Value* pOffset = (offset == 0) ?
+                               pBaseIndex :
+                               m_pBuilder->CreateAdd(pBaseIndex, m_pBuilder->getInt32(offset));
 
-            Type* pIntLoadType = nullptr;
-            Type* pFloatLoadType = nullptr;
+        Type* pIntLoadType = nullptr;
+        uint32_t loadSize;
 
+        // Load the greatest possible size
+        if (alignment >= 4 && remainingBytes >= 4)
+        {
             if (remainingBytes >= 16)
             {
                 pIntLoadType = VectorType::get(Type::getInt32Ty(*m_pContext), 4);
-                pFloatLoadType = VectorType::get(Type::getFloatTy(*m_pContext), 4);
-                remainingBytes -= 16;
+                loadSize = 16;
             }
-            else if (remainingBytes >= 8)
+            else if (remainingBytes >= 12 && !isInvariant)
             {
-                pIntLoadType = VectorType::get(Type::getInt32Ty(*m_pContext), 2);
-                pFloatLoadType = VectorType::get(Type::getFloatTy(*m_pContext), 2);
-                remainingBytes -= 8;
+                pIntLoadType = VectorType::get(Type::getInt32Ty(*m_pContext), 3);
+                loadSize = 12;
             }
-            else if (remainingBytes >= 4)
+            else if (remainingBytes >= 8)
             {
-                pIntLoadType = Type::getInt32Ty(*m_pContext);
-                pFloatLoadType = Type::getFloatTy(*m_pContext);
-                remainingBytes -= 4;
+                pIntLoadType = VectorType::get(Type::getInt32Ty(*m_pContext), 2);
+                loadSize = 8;
             }
             else
             {
-                LLPC_NEVER_CALLED();
+                // remainingBytes >= 4
+                pIntLoadType = Type::getInt32Ty(*m_pContext);
+                loadSize = 4;
             }
+        }
+        else if (alignment >= 2 && remainingBytes >= 2)
+        {
+            pIntLoadType = Type::getInt16Ty(*m_pContext);
+            loadSize = 2;
+        }
+        else
+        {
+            pIntLoadType = Type::getInt8Ty(*m_pContext);
+            loadSize = 1;
+        }
 
-            Value* pPartLoad = nullptr;
+        Value* partLoad = nullptr;
```

Comment:
partLoad -> pPartLoad
This is a coding style change. Please drop it. Keep current style. Tim will do this with dedicated commits in the future.